### PR TITLE
feature: add stackchart render and add dice_issue_state_circulation

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR

Add one of the following kinds:
/kind feature

#### What this PR does / why we need it:
feature: add stackchart render and add dice_issue_state_circulation

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=263764&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sInN0YXRlcyI6WzQ0MDIsNzEwNCw3MTA1LDQ0MDMsNDQwNCw3MTA2LDQ0MDYsNDQwNyw0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl0sImFzc2lnbmVlSURzIjpbIjEwMDEyNjEiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=680&type=TASK)

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: .erda/migrations/cmdb/20211214-issue-state-circulation.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE `dice_issue_state_circulation`
```
Comment (Effet) [inline .erda/migrations/cmdb/20211214-issue-state-circulation.sql:1]: erda_issue_state_circulation

File: .erda/migrations/cmdb/20211214-issue-state-circulation.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+(
+    `id`         bigint(20) NOT NULL AUTO_INCREMENT COMMENT '主键',
```
Comment (Effet) [inline .erda/migrations/cmdb/20211214-issue-state-circulation.sql:3]: id should be UUID, I mean, varchar

File: modules/dop/dao/issue_state_circulation.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	ProjectID uint64
+	IssueID   uint64
+	StateFrom uint64
+	StateTo   uint64
+	Creator   string
+}
+
+func (IssueStateCirculation) TableName() string {
+	return "dice_issue_state_circulation"
```
Comment (Effet) [inline modules/dop/dao/issue_state_circulation.go:32]: erda_issue_state_circulation

Comment (littlejiancc) [reply modules/dop/dao/issue_state_circulation.go:32]: The others are prefixed with `dice` like issue,issue_state and so on

File: .erda/migrations/cmdb/20211214-issue-state-circulation.sql (hunk 1 above)
Comment (littlejiancc) [reply .erda/migrations/cmdb/20211214-issue-state-circulation.sql:1]: The others are prefixed with dice like issue,issue_state and so on

Comment (littlejiancc) [reply .erda/migrations/cmdb/20211214-issue-state-circulation.sql]: done

File: .erda/migrations/cmdb/20211214-issue-state-circulation.sql (hunk 2 above)
Comment (littlejiancc) [reply .erda/migrations/cmdb/20211214-issue-state-circulation.sql]: done

File: modules/dop/dao/issue_state_circulation.go (hunk 3 above)
Comment (littlejiancc) [reply modules/dop/dao/issue_state_circulation.go]: done

File: .erda/migrations/cmdb/20211214-issue-state-circulation.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    `id`         varchar(36)  NOT NULL COMMENT 'primary key',
+    `created_at` datetime     NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '表记录创建时间',
+    `updated_at` datetime     NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT '表记录更新时间',
+    `project_id` bigint(20) NOT NULL DEFAULT '0' COMMENT '所属项目 ID',
+    `state_from` bigint(20) NOT NULL DEFAULT '0' COMMENT '更新前状态',
+    `state_to`   bigint(20) NOT NULL DEFAULT '0' COMMENT '更新后状态',
+    `creator`    varchar(255) NOT NULL DEFAULT '' COMMENT '创建人',
+    `issue_id`   bigint(20) NOT NULL DEFAULT '0' COMMENT 'issue_id',
+    PRIMARY KEY (`id`),
+    KEY          `idx_project_id` (`project_id`) USING BTREE
```
Comment (Effet) [inline .erda/migrations/cmdb/20211214-issue-state-circulation.sql:12]: why BTREE?

File: .erda/migrations/cmdb/20211214-issue-state-circulation.sql [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+) ENGINE=InnoDB AUTO_INCREMENT=6432 DEFAULT CHARSET=utf8mb4 COMMENT='issue状态流转表';
```
Comment (Effet) [inline .erda/migrations/cmdb/20211214-issue-state-circulation.sql:13]: should not specify AUTO_INCREMENT

File: .erda/migrations/cmdb/20211214-issue-state-circulation.sql [hunk 6]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE `erda_issue_state_circulation`
```
Comment (Effet) [inline .erda/migrations/cmdb/20211214-issue-state-circulation.sql:1]: transition is better than circulation?

File: modules/dop/component-protocol/components/requirement-task-overview/distributePage/topFilter/render.go [hunk 7]
```diff
@@ -238,7 +238,7 @@ func (f *ComponentFilter) getPropIterationsOptions() (map[int64]apistructs.Itera
 			return &t
 		}(),
 		FinishedAt: func() *time.Time {
-			t := time.Now().AddDate(0, -1, 0)
+			t := time.Now().AddDate(0, 0, 0)
```
Comment (Effet) [inline modules/dop/component-protocol/components/requirement-task-overview/distributePage/topFilter/render.go:241]: what's the mean of add zero?

File: modules/dop/dao/issue.go [hunk 8]
```diff
@@ -1020,3 +1020,9 @@ func (client *DBClient) FindIssueChildrenTimeRange(id uint64) (*time.Time, *time
 	}
 	return res.Min, res.Max, nil
 }
+
+func (client *DBClient) ListIssueForMigration() ([]Issue, error) {
```
Comment (Effet) [inline modules/dop/dao/issue.go:1024]: for migration what?

**Review by Effet — CHANGES_REQUESTED** [review]

File: .erda/migrations/cmdb/20211214-issue-state-circulation.sql (hunk 5 above)
Comment (littlejiancc) [reply .erda/migrations/cmdb/20211214-issue-state-circulation.sql:13]: done

File: .erda/migrations/cmdb/20211214-issue-state-circulation.sql (hunk 6 above)
Comment (littlejiancc) [reply .erda/migrations/cmdb/20211214-issue-state-circulation.sql:1]: done

File: modules/dop/component-protocol/components/requirement-task-overview/distributePage/topFilter/render.go (hunk 7 above)
Comment (littlejiancc) [reply modules/dop/component-protocol/components/requirement-task-overview/distributePage/topFilter/render.go:241]: remove it

File: modules/dop/dao/issue.go (hunk 8 above)
Comment (littlejiancc) [reply modules/dop/dao/issue.go:1024]: done

**Review by Effet — APPROVED** [review]

**Pull request merged**